Automattic/woocommerce-payments
# Update mock_wcpay_request to properly return the mock response

Fixes #

I noticed this issue when trying to switch the approach for my own question here https://github.com/Automattic/woocommerce-payments/pull/5879/files#r1153037010

#### Changes proposed in this Pull Request

The current `mock_wcpay_request` method is just mocking the response for `WC_Payments_API_Client::send_request`. 

To use the response from this request, I still need to mock `format_response` even my request concrete class does not extend `format_response`. 

With this change, I just use: 

```php
$mock_response = [
   // something here.
]; 
$this->wcpay_mock_request( Concrete_Request::class, 1, null, $mock_response);
```

Otherwise, I need to do something like, which is both verbose and not what I am expecting from reading [the doc](https://github.com/Automattic/woocommerce-payments/blob/547a0d2bbbfc3f505077cf051c8c669b71eb34a9/includes/core/server/CONTRIBUTING.md#testing) as `$mock_response` does not give me the expected result when calling `mock_wcpay_request`:

```php
$mock_response = [
	// something here.
];
$request = $this->wcpay_mock_request( Concrete_Request::class, 1);
$request->expect( 'format_response' )
	->willReturn ( new Response( $mock_response ) ); 
``` 

Also the doc mention that I just need to mock `format_response` in case I have custom `format_response` method.









#### Testing instructions





* No specific instruction for this, ideally no current tests fails. 
* Focus on my reasoning and see if it makes sense to change. 
* It's OK to discard this PR if we have a better way to get the same thing. Note that:  I am lacking of the full knowledge from the phase 1 of WCPay as Platform.

-------------------


**Post merge**
